qiskit_nature/algorithms/initial_points/mp2_initial_point.py

```diff
@@ -0,0 +1,363 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Second-order Møller-Plesset perturbation theory (MP2) computation."""
+
+from __future__ import annotations
+
+import numpy as np
+from qiskit_nature.exceptions import QiskitNatureError
+
+from qiskit_nature.properties.second_quantization.electronic import ElectronicEnergy
+from qiskit_nature.properties.second_quantization.electronic.bases import ElectronicBasis
+from qiskit_nature.circuit.library import UCC
+from qiskit_nature.properties.second_quantization.electronic.integrals.electronic_integrals import (
+    ElectronicIntegrals,
+)
+from qiskit_nature.properties.second_quantization.second_quantized_property import (
+    GroupedSecondQuantizedProperty,
+)
+
+from .initial_point import InitialPoint
+
+
+class MP2InitialPoint(InitialPoint):
+    """Second-order Møller-Plesset perturbation theory (MP2) computation.
+
+    A class for computing the second-order Møller-Plesset perturbation theory (MP2) corrections.
+    This is intended for use as an initial point for the VQE parameters in combination with a UCC
+    ansatz.
+
+    :class:`MP2InitialPoint` requires two-body molecular orbital electronic integrals and orbital
+    energies from
+    :class:`~qiskit_nature.properties.second_quantization.electronic.ElectronicEnergy`,
+    which should be passed in via the :attr:`grouped_property` attribute.
+
+    :class:`MP2InitialPoint` also requires the :attr:`excitation list` from the :attr:`~ansatz`
```

Comment:
```suggestion
    :class:`MP2InitialPoint` also requires the :attr:`~excitation_list` from the :attr:`~ansatz`
```
Assuming the `~` is really needed since it is also used below.

Reply:
Follow-up: if the `~` is indeed needed, you probably also need it for the `grouped_property` in various other places of this docstring.

Reply:
Good spot. It looks like it does format the same in the docs with and without the tilde (once the underscore is fixed). I'll remove the tildes for consistency and check that all the cross-references work.
<img width="812" alt="Screen Shot 2022-04-22 at 10 56 25" src="https://user-images.githubusercontent.com/14238050/164683815-ff816535-84e1-42bb-9b4c-e7fa81539a56.png">
<img width="822" alt="Screen Shot 2022-04-22 at 10 56 32" src="https://user-images.githubusercontent.com/14238050/164683822-6c0722df-68b2-434f-8dfc-54b7a6f0b75b.png">